tensorflow/core/kernels/data/name_utils.h

```diff
@@ -27,39 +29,65 @@ namespace name_utils {
 extern const char kDelimiter[];
 extern const char kDefaultDatasetDebugStringPrefix[];
 
+struct OpNameParams {
+  OpNameParams() = default;
+
+  explicit OpNameParams(int op_version) : op_version(op_version){};
```

Comment:
I would prefer not to have a constructor that takes arguments because that becomes outdated the moment we add an argument

this means that the params have to be setup using multiple statements (you could use the `WithFoo()` to avoid that) but that's a small price to pay for not having to either update all the callers every time a new parameter is introduced

the callers should do:

```
name_utils::OpNameParams params;
... // explicitly set all the non-default parameters one by one here
```